programs/clearing_house/src/controller/pnl.rs

```diff
@@ -124,3 +137,114 @@ pub fn settle_pnl(
 
     Ok(())
 }
+
+pub fn settle_expired_position(
+    market_index: u64,
+    user: &mut User,
+    user_key: &Pubkey,
+    market_map: &MarketMap,
+    bank_map: &BankMap,
+    oracle_map: &mut OracleMap,
+    now: i64,
+    fee_structure: &FeeStructure,
+) -> ClearingHouseResult {
+    {
+        let bank = &mut bank_map.get_quote_asset_bank_mut()?;
+        update_bank_cumulative_interest(bank, now)?;
+    }
+
+    settle_funding_payment(
+        user,
+        user_key,
+        market_map.get_ref(&market_index)?.deref(),
+        now,
+    )?;
+
+    // cannot settle pnl this way on a user who is in liquidation territory
+    if !(meets_maintenance_margin_requirement(user, market_map, bank_map, oracle_map)?) {
```

Comment:
how does a user being bankrupt affect settlement?

Reply:
added rust test in `delisting.ts`